src/client/components/pages/parts/add-entity-to-collection-modal.js

```diff
@@ -0,0 +1,217 @@
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as bootstrap from 'react-bootstrap';
+import {lowerCase, uniqBy} from 'lodash';
+import EntitySearchFieldOption from '../../../entity-editor/common/entity-search-field-option';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {genEntityIconHTMLElement} from '../../../helpers/entity';
+import request from 'superagent';
+
+
+const {Alert, Badge, Button, ButtonGroup, Col, Modal} = bootstrap;
+
+class AddEntityToCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			entities: [],
+			error: null
+		};
+
+		this.handleAddEntity = this.handleAddEntity.bind(this);
+		this.handleRemoveEntity = this.handleRemoveEntity.bind(this);
+		this.handleChangeEntity = this.handleChangeEntity.bind(this);
+		this.getCleanedEntities = this.getCleanedEntities.bind(this);
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.handleAlertDismiss = this.handleAlertDismiss.bind(this);
+	}
+
+	getCleanedEntities() {
+		return uniqBy(this.state.entities.filter(entity => entity && entity.id !== null), 'id');
+	}
+
+	handleAddEntity() {
+		this.setState(prevState => ({
+			entities: [...prevState.entities, {id: null, name: ''}]
+		}));
+	}
+
+	handleChangeEntity(newEnt, index) {
+		let newEntity;
+		if (!newEnt) {
+			newEntity = {
+				id: null,
+				name: ''
+			};
+		}
+		else {
+			newEntity = newEnt;
+		}
+
+		this.setState((prevState) => {
+			const newEntities = prevState.entities;
+			newEntities[index] = newEntity;
+			return {
+				entities: newEntities
+			};
+		});
+	}
+
+	handleRemoveEntity(index) {
+		this.setState(prevState => ({
+			entities: prevState.entities.splice(index, 1) && prevState.entities
+		}));
+	}
+
+	handleAlertDismiss() {
+		this.setState({error: null});
+	}
+
+	handleSubmit() {
+		const cleanedEntities = this.getCleanedEntities();
+		const bbids = cleanedEntities.map(entity => entity.id);
+		if (bbids.length) {
+			request.post(`/collection/${this.props.collectionId}/add`)
+				.send({bbids})
+				.then((res) => {
+					window.location.href = `/collection/${this.props.collectionId}`;
+				}, (error) => {
+					this.setState({
+						error: 'Something went wrong! Please try again later'
+					});
+				});
+		}
+		else {
+			this.setState({
+				error: `No ${lowerCase(this.props.collectionType)} selected`
+			});
+		}
+	}
+
+	render() {
+		if (this.state.entities.length === 0) {
+			this.handleAddEntity();
+		}
+
+		let errorComponent = null;
+		if (this.state.error) {
+			errorComponent =
+				(
+					<div className="text-center margin-top-1">
+						<Alert bsStyle="danger" onDismiss={this.handleAlertDismiss}>{this.state.error}</Alert>;
```

Comment:
The trailing semicolon ';' at the end appears as text in the page, right after the alert:

<img width="1170" alt="Capture d’écran 2020-08-11 à 13 00 17" src="https://user-images.githubusercontent.com/6179856/89889852-a967f900-dbd2-11ea-8223-e6710339163a.png">